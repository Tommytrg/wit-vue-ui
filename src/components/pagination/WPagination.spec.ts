import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import WPagination from './WPagination.vue'

describe('WPagination', () => {
  it('renders properly without divider', () => {
    const wrapper = mount(WPagination, { props: { total: 100, pageSize: 5, page: 1 } })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders properly with divider', () => {
    const wrapper = mount(WPagination, { props: { total: 10, pageSize: 5, page: 4 } })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('click on page changes current page', async () => {
    const wrapper = mount(WPagination, { props: { total: 10, pageSize: 5, page: 1 } })

    const buttons = wrapper.findAll('button')

    expect(buttons[2].element.textContent).toBe('2')

    await buttons[2].trigger('click')

    const selected = wrapper.find('.bg-wit-blue-500')

    expect(selected.text()).toBe('2')
  })

  it('click on previous changes current page', async () => {
    const wrapper = mount(WPagination, { props: { total: 10, pageSize: 5, page: 2 } })

    const buttons = wrapper.findAll('button')

    expect(buttons[0].element.textContent).toBe('Previous')

    await buttons[0].trigger('click')

    const selected = wrapper.find('.bg-wit-blue-500')

    expect(selected.text()).toBe('1')
  })

  it('click on next changes current page', async () => {
    const wrapper = mount(WPagination, { props: { total: 10, pageSize: 5, page: 1 } })

    const buttons = wrapper.findAll('button')

    expect(buttons[3].element.textContent).toBe('Next')

    await buttons[3].trigger('click')

    const selected = wrapper.find('.bg-wit-blue-500')

    expect(selected.text()).toBe('2')
  })
})
