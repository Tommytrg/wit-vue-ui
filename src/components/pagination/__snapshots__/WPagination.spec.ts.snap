// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WPagination > renders properly with divider 1`] = `
<nav
  aria-label="Pagination"
  class="flex items-center gap-xs"
>
  <button
    aria-label="Previous"
    class="min-h-[38px] min-w-[38px] py-sm px-sm inline-flex justify-center items-center gap-sm text-sm rounded-lg hover:bg-wit-blue-400 focus:outline-none"
    type="button"
  >
    <svg
      class="shrink-0 size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m15 18-6-6 6-6"
      />
    </svg>
    <span>
      Previous
    </span>
  </button>
  <div
    class="flex items-center gap-x-xs"
  >
    
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      1
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      2
    </button>
    
  </div>
  <button
    aria-label="Next"
    class="min-h-lg min-w-lg py-sm px-sm inline-flex justify-center items-center gap-xs text-sm rounded-lg hover:bg-wit-blue-400 focus:outline-none opacity-10 pointer-events-none"
    disabled=""
    type="button"
  >
    <span>
      Next
    </span>
    <svg
      class="shrink-0 size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m9 18 6-6-6-6"
      />
    </svg>
  </button>
</nav>
`;

exports[`WPagination > renders properly without divider 1`] = `
<nav
  aria-label="Pagination"
  class="flex items-center gap-xs"
>
  <button
    aria-label="Previous"
    class="opacity-50 pointer-events-none min-h-[38px] min-w-[38px] py-sm px-sm inline-flex justify-center items-center gap-sm text-sm rounded-lg hover:bg-wit-blue-400 focus:outline-none"
    disabled=""
    type="button"
  >
    <svg
      class="shrink-0 size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m15 18-6-6 6-6"
      />
    </svg>
    <span>
      Previous
    </span>
  </button>
  <div
    class="flex items-center gap-x-xs"
  >
    
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg outline-none bg-wit-blue-500 pointer-events-none hover:bg-wit-blue-400"
      type="button"
    >
      1
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      2
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      3
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      4
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      5
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg"
      disabled=""
      type="button"
    >
      ...
    </button>
    <button
      class="min-h-[38px] min-w-[38px] flex justify-center items-center py-sm px-sm text-sm rounded-lg hover:bg-wit-blue-400"
      type="button"
    >
      20
    </button>
    
  </div>
  <button
    aria-label="Next"
    class="min-h-lg min-w-lg py-sm px-sm inline-flex justify-center items-center gap-xs text-sm rounded-lg hover:bg-wit-blue-400 focus:outline-none"
    type="button"
  >
    <span>
      Next
    </span>
    <svg
      class="shrink-0 size-3.5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m9 18 6-6-6-6"
      />
    </svg>
  </button>
</nav>
`;
